force-app/main/default/classes/SOQL.cls

```diff
@@ -130,9 +159,13 @@ public with sharing class SOQL {
      * @return A list of Lead records.
      * Update the corresponding test class and testGetLeadsBySourceOrTitle method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}    
+    public static List<Lead> getLeadsBySourceOrTitle(String leadSource, String keyword) {
+        // Query for leads based on the given Lead Source or keyword
+        String query = 'SELECT Id, Name, Title, LastName, Company, Phone, Email, LeadSource FROM Lead WHERE LeadSource = :leadSource OR Title LIKE :keyword';
+        List<Lead> leads = Database.query(query);
+
```

Comment:
really cool to see you experiment with dynamic (string) queries.

However, in practice, if we don't need to write pieces of a query dynamically, I want to make sure you know that it's better to keep them as static unchanging queries.

Love the exploration effort.